bridge/discord/transmitter/transmitter.go

```diff
@@ -0,0 +1,247 @@
+	if err != nil {
+		return nil, fmt.Errorf("could not execute existing webhook: %w", err)
+	}
+
+	return msg, nil
+}
+
+// Edit will edit a message in a channel, if possible.
+func (t *Transmitter) Edit(channelID string, messageID string, params *discordgo.WebhookParams) error {
+	wh := t.getWebhook(channelID)
+
+	if wh == nil {
+		return ErrWebhookNotFound
+	}
+
+	uri := discordgo.EndpointWebhookToken(wh.ID, wh.Token) + "/messages/" + messageID
+	_, err := t.session.RequestWithBucketID("PATCH", uri, params, discordgo.EndpointWebhookToken("", ""))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// HasWebhook checks whether the transmitter is using a particular webhook.
+func (t *Transmitter) HasWebhook(id string) bool {
+	t.mutex.RLock()
+	defer t.mutex.RUnlock()
+
+	for _, wh := range t.channelWebhooks {
+		if wh.ID == id {
+			return true
+		}
+	}
+
+	return false
+}
+
+// AddWebhook allows you to register a channel's webhook with the transmitter.
+func (t *Transmitter) AddWebhook(channelID string, webhook *discordgo.Webhook) (replaced bool) {
+	t.Log.Debugf("Manually added webhook %#v to channel %#v", webhook.ID, channelID)
+	t.mutex.Lock()
+	defer t.mutex.Unlock()
+
+	_, replaced = t.channelWebhooks[channelID]
+	t.channelWebhooks[channelID] = webhook
+	return replaced
+}
+
+// RefreshGuildWebhooks loads "relevant" webhooks into the transmitter, with careful permission handling.
+//
+// Notes:
+// - A webhook is "relevant" if it was created by this bot -- the ApplicationID should match the bot's ID.
+// - The term "having permission" means having the "Manage Webhooks" permission. See ErrPermissionDenied for more information.
+// - This function is additive and will not unload previously loaded webhooks.
+// - A nil channelIDs slice is treated the same as an empty one.
+//
+// If the bot has guild-wide permission:
+// 1. it will load any "relevant" webhooks from the entire guild
+// 2. the given slice is ignored
+//
+// If the bot does not have guild-wide permission:
+// 1. it will load any "relevant" webhooks in each channel
+// 2. a single error will be returned if any error occurs (incl. if there is no permission for any of these channels)
+//
+// If any channel has more than one "relevant" webhook, it will randomly pick one.
+func (t *Transmitter) RefreshGuildWebhooks(channelIDs []string) error {
+	t.Log.Debugln("Refreshing guild webhooks")
+
+	botID, err := getDiscordUserID(t.session)
+	if err != nil {
+		return fmt.Errorf("could not get current user: %w", err)
+	}
+
+	// Get all existing webhooks
+	hooks, err := t.session.GuildWebhooks(t.guild)
+	manuallyFetchHooks := len(channelIDs) > 0
+
+	// Check to make sure we have permissions
+	if isDiscordPermissionError(err) {
+		// If we shouldn't try manually fetching hooks, return an error
+		if !manuallyFetchHooks {
+			return fmt.Errorf("the 'Manage Webhooks' permission is required: %w", err)
+		}
+	} else if err != nil {
+		return fmt.Errorf("could not get webhooks: %w", err)
+	}
```

Comment:
```suggestion
if err != nil {
        // Check to make sure we have permissions
        // If we shouldn't try manually fetching hooks, return an error
	if isDiscordPermissionError(err) && !manuallyFetchHooks {
		return fmt.Errorf("the 'Manage Webhooks' permission is required: %w", err)
	}

	return fmt.Errorf("could not get webhooks: %w", err)
}
```

Reply:
this makes it more readable

Reply:
I'll try something like this, this won't immediately work, as we don't want to return an error when `isDiscordPermissionError(err)` and `manuallyFetchHooks` is true.

(This is a global permission error, and tries per-channel permissions as a fallback)